includes/validation/class-amp-validation-error-taxonomy.php

```diff
@@ -384,29 +446,54 @@ public static function get_validation_error_count( $args = array() ) {
 	 */
 	public static function filter_posts_where_for_validation_error_status( $where, WP_Query $query ) {
 		global $wpdb;
-		if (
-			in_array( AMP_Invalid_URL_Post_Type::POST_TYPE_SLUG, (array) $query->get( 'post_type' ), true )
-			&&
-			is_numeric( $query->get( self::VALIDATION_ERROR_STATUS_QUERY_VAR ) )
-		) {
-			$where .= $wpdb->prepare(
-				" AND (
-					SELECT 1
-					FROM $wpdb->term_relationships
-					INNER JOIN $wpdb->term_taxonomy ON $wpdb->term_taxonomy.term_taxonomy_id = $wpdb->term_relationships.term_taxonomy_id
-					INNER JOIN $wpdb->terms ON $wpdb->terms.term_id = $wpdb->term_taxonomy.term_id
-					WHERE
-						$wpdb->term_taxonomy.taxonomy = %s
-						AND
-						$wpdb->term_relationships.object_id = $wpdb->posts.ID
-						AND
-						$wpdb->terms.term_group = %d
-					LIMIT 1
-				)",
-				self::TAXONOMY_SLUG,
-				$query->get( self::VALIDATION_ERROR_STATUS_QUERY_VAR )
-			);
+
+		// If the post type is not correct, return the $where clause unchanged.
+		if ( ! in_array( AMP_Invalid_URL_Post_Type::POST_TYPE_SLUG, (array) $query->get( 'post_type' ), true ) ) {
+			return $where;
+		}
+
+		$error_status = $query->get( self::VALIDATION_ERROR_STATUS_QUERY_VAR );
+		$error_type   = $query->get( self::VALIDATION_ERROR_TYPE_QUERY_VAR );
+
+		/*
+		 * Selecting the 'All Statuses' <option> sends a value of '-1' to indicate that this should not filter.
+		 * So ensure that the value is_numeric(), and is not '-1'.
+		 */
+		$is_error_status_present = is_numeric( $error_status ) && self::NO_FILTER_VALUE !== intval( $error_status );
+		$is_error_type_present   = in_array( $error_type, array( self::HTML_ELEMENT_ERROR_TYPE, self::HTML_ATTRIBUTE_ERROR_TYPE, self::JS_ERROR_TYPE, self::CSS_ERROR_TYPE ), true );
+
+		// If neither the error status nor the type is present, there is no need to filter the $where clause.
+		if ( ! $is_error_status_present && ! $is_error_type_present ) {
+			return $where;
 		}
+
+		// If there is no valid error status query var, use the wildcard so that any status will match.
+		$error_status = $is_error_status_present ? $error_status : '*';
+
+		// If there is no valid error type query var, use %s so that anything will match.
+		$error_type = $is_error_type_present ? '%"type":"' . $wpdb->esc_like( $error_type ) . '"%' : '%';
+
+		$where .= $wpdb->prepare(
+			" AND (
+				SELECT 1
+				FROM $wpdb->term_relationships
+				INNER JOIN $wpdb->term_taxonomy ON $wpdb->term_taxonomy.term_taxonomy_id = $wpdb->term_relationships.term_taxonomy_id
+				INNER JOIN $wpdb->terms ON $wpdb->terms.term_id = $wpdb->term_taxonomy.term_id
+				WHERE
+					$wpdb->term_taxonomy.taxonomy = %s
+					AND
+					$wpdb->term_relationships.object_id = $wpdb->posts.ID
+					AND
+					$wpdb->terms.term_group = %s
+					AND
+					$wpdb->term_taxonomy.description LIKE %s
```

Comment:
This addition to the `WHERE` clause is the same as before, except for the 2 lines above:

```php
AND
$wpdb->term_taxonomy.description LIKE %s
```

Reply:
Maybe it'd be better to search the `amp_invalid_url` post content for the correct `error_type`, instead of searching the `amp_validation_error` term `description`, like it does here.

I think that's what [you suggested here](https://github.com/Automattic/amp-wp/issues/1363#issuecomment-417191346).

Reply:
Yeah, but what you've done here seems fine as well.